src/nfa.ts

```diff
@@ -117,6 +116,73 @@ export class NFA implements ReadonlyNFA {
 		return match(0, nodes.initial);
 	}
 
+	// Thompson simulation NFA matching from Cox article
+	match(input: string) {
+		interface ThompsonList {
+			states: Map<NFA.Node, null>; // store character leading to state
```

Comment:
You probably wanted to use a [`Set<NFA.State>`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set) here.